examples/pem/pem.c

```diff
@@ -918,12 +930,14 @@ int main(int argc, char* argv[])
         else if ((strcmp(argv[0], "-?") == 0) ||
                  (strcmp(argv[0], "--help") == 0)) {
             Usage();
-            return 0;
+            ret = 0;
+            goto out;
         }
         else {
             fprintf(stderr, "Bad option: %s\n", argv[0]);
             Usage();
-            return 1;
+            ret = 1;
+            goto out;
```

Comment:
Change code to not require goto.

Reply:
Can't justify disrupting all the incumbent code in the example just to refactor for that.  Besides, `goto out` is legal under MISRA C 2012.  The incumbent code made no attempt to clean up its allocations on error -- just `return`ed -- and we're just retrofitting cleanup here with a minimum of fuss.

Btw I don't have a clue why `main()` in that file is set up to `return` rather than `exit` -- I'm assuming something else uses it as an app-callable function somehow.  If it is in fact only used as a standalone application, all those `return`s, now `goto out`s, can be `exit()`s.

Reply:
Don't care, I care about the fact that I had it working without the need for gotos before.
I'll remove them after this PR has been merged then.

Reply:
Oh, you're replacing the return's.
Fine I'll take the time to fix that later.